### PR TITLE
fix(cli): support running manager-api cmd in non-default diretory

### LINKED ISSUE #1147: bug: manager-api -p doesn't work
# Bug report

## Describe the bug

manager-api `-p` parameter doesn't work, it will prompt  `panic: fail to read configuration: ./conf/conf.yaml`

1. move `output` to another dir
2. use `ln -s` create manager-api soft links to `/usr/bin`
3. use manager-api -p `/path/to/output`
4. it will prompt  `panic: fail to read configuration: ./conf/conf.yaml`

@nic-chen

### PR DESCRIPTION
**Pull request by imjoey**
Signed-off-by: imjoey <majunjiev@gmail.com>

Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

Fixes #1147 .
___
### Bugfix
- Description

Running the prebuilt `manager-api` under non-default directory may lead to failed to start, see #1147 for details. This is mainly caused by the `-p` flag of `manager-api` is not worked as expected.

- How to fix?

Introduce a new value `test` for `ENV` environment to indicates the program is running a unit testing. If `ENV` is set to `test`, the original `init()` functions are retained, otherwise we explicitly calling `InitConf()` and `InitLogger()` functions to  initialization, right after receiving the input `WorkDir` parameter via `-p` flag. This would ensure configs and logger could be initialized just at the right time.

### CODE REVIEW

**Comment by juzhiyuan** [comment]
very quick...

**Comment by juzhiyuan** [comment]
Just ignored this check, it failed due to we only have 1 commit in this PR, once we pushed another, it will success. Not required.

image

**Comment by imjoey** [comment]
> Just ignored this check, it failed due to we only have 1 commit in this PR, once we pushed another, it will success. Not required.
> 
> image

@juzhiyuan Got you~ Thanks.

**Comment by imjoey** [comment]
CI passed and it's ready for review. ping @nic-chen @tokers , much appreciated.

**Comment by nic-chen** [comment]
Thanks. We should add test case for it.

File: api/internal/conf/conf.go [hunk 1]
```diff
@@ -100,9 +101,11 @@ type Config struct {
 	Authentication Authentication
 }
 
-// TODO: it is just for integration tests, we should call "InitLog" explicitly when remove all handler's integration tests
+// ENV=test is for integration tests only, other ENV should call "InitConf" explicitly
```
Comment (nic-chen) [inline api/internal/conf/conf.go:104]: TODO should be retained here. We need to finally remove `init()`.

Comment (imjoey) [reply api/internal/conf/conf.go:105]: > TODO should be retained here. We need to finally remove `init()`.

@nic-chen Done, thanks.

**Comment by starsz** [comment]
Hi @imjoey, can you add some tests for it?

**Comment by imjoey** [comment]
> Hi @imjoey, can you add some tests for it?

@starsz sure, I will add soon. Thanks.

**Comment by imjoey** [comment]
@nic-chen @starsz Test case added and ready for review. Any feedback is appreciated. Thanks.

File: api/test/shell/cli_test.sh [hunk 2]
```diff
@@ -126,6 +126,28 @@ if [[ $res != "hi~" ]]; then
 fi
 clean_up
 
+# run with -p flag out of the default directory
+workDir=$(pwd)
```
Comment (nic-chen) [inline api/test/shell/cli_test.sh:130]: Looks like it's still in default dir. could we mv it to another dir to test it ?

Comment (imjoey) [reply api/test/shell/cli_test.sh:130]: @nic-chen From my point of view, the default directory only makes sense for `manager-api` binary. But it's indeed not intuitive. I would renew this test case. Thanks.

Comment (nic-chen) [reply api/test/shell/cli_test.sh:130]: oh, yes, that make sense.

**Review by nic-chen — APPROVED** [review]

File: api/test/shell/cli_test.sh (hunk 2 above)
Comment (imjoey) [reply api/test/shell/cli_test.sh:130]: @nic-chen Hi, PR renewed with a more intuitive way. Thanks for reviewing. 😄

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**